pyfar/dsp/window.py

```
@@ -0,0 +1,378 @@
+        'bartlett', 'kaiser', 'kaiserBessel' , 'flattop' or 'dolphChebychev'
+    
+    windowLength: int
+        sets window length in samples of selected window
+        
+    windowStartIndex: int
+        sets the start index! of the window
+        
+    Returns
+    -------
+    out : pyfar
+        window coefficients
+    """
+    
+    #check input and default values
+    if not isinstance(signal, Signal):
+        raise TypeError('Input data has to be of type: Signal.')
+        
+    if not isinstance(windowType, str):
+        raise TypeError('''windowType has to be a string of the following:
+                        'rect', 'hann', 'hamming', 'blackman', 'bartlett', 'kaiser',
+                        'kaiserBessel' , 'flattop' or 'dolphChebychev'.''')
+    if not isinstance(windowLength,int):
+        raise TypeError('windowLength has to be of type int.')
+        
+    if not isinstance(windowStartIndex,int) or windowStartIndex < 0:
+        raise TypeError('windowStartIndex has to be of type int and >= 0 .')
+    
+    #create selected window with simulated switch case   
+    switcher = {
+        'rect':rect,
+        'hann':hann,
+        'hamming':hamming,
+        'blackman':blackman,
+        'bartlett':bartlett,
+        'kaiser':kaiser,
+        'kaiserBessel':kaiserBessel,
+        'flattop':flattop,
+        'dolphChebychev':dolphChebychev,
+        }
+    if windowType in switcher:
+        win = switcher.get(windowType)(windowLength)
+    else:
+        raise NameError('''windowType has to be a sting of the following:
+                        'rect', 'hann', 'hamming', 'blackman', 'bartlett', 'kaiser',
+                        'kaiserBessel' , 'flattop' or 'dolphChebychev'.''')
+    #%% apply windowing to time domain copy of signal
+    signalCopy = signal.copy()
+    signalCopyFlat= signalCopy.flatten()
+    if windowLength > np.size(signalCopyFlat.time[0]):
+        print('>>>>>>> window is longer than signal!')
```

you can use `signal.n_samples` for the check and you should raise a `ValueError` instead of the print.